codegenerator/cli/src/config_parsing/system_config.rs

```diff
@@ -531,6 +531,44 @@ impl SystemConfig {
         let mut networks: NetworkMap = HashMap::new();
         let mut contracts: ContractMap = HashMap::new();
 
+        // Create a new ParsedProjectPaths that uses the output field from config if specified
+        let final_project_paths = {
+            let output_path = match &human_config {
+                HumanConfig::Evm(evm_config) => evm_config.output.as_ref(),
+                HumanConfig::Fuel(fuel_config) => fuel_config.output.as_ref(),
+            };
+
+            match output_path {
+                Some(output) => {
+                    // If output is specified, create a new ParsedProjectPaths with the custom output path
+                    // The output path is relative to the config file location
+                    let config_dir = project_paths.config.parent().ok_or_else(|| {
+                        anyhow!("Unexpected config file should have a parent directory")
+                    })?;
+
+                    let output_relative_path = PathBuf::from(output);
+                    if let Some(Component::ParentDir) =
+                        output_relative_path.components().peekable().peek()
+                    {
+                        return Err(anyhow!("Output folder must be in project directory"));
```

Comment:
btw there is a shorthand for this
```suggestion
                        anyhow::bail!("Output folder must be in project directory");
```